pkg/middleware/csrf.go

```diff
@@ -34,16 +35,30 @@ func CSRF(loginCookieName string, logger log.Logger) func(http.Handler) http.Han
 				http.Error(w, err.Error(), http.StatusBadRequest)
 				return
 			}
-
 			origin, err := url.Parse(r.Header.Get("Origin"))
 			if err != nil {
 				logger.Error("error parsing Origin header", "err", err)
 			}
-			if err != nil || netAddr.Host == "" || (origin.String() != "" && origin.Hostname() != netAddr.Host) {
-				http.Error(w, "origin not allowed", http.StatusForbidden)
-				return
+			// env var GF_USE_BEHIND_PROXY controls if X-Forwarded-Host is used or not
+			if UseBehindProxy := os.Getenv("GF_USE_BEHIND_PROXY"); UseBehindProxy == "true" {
+				// X-Forwarded-Host for (reverse) proxy scenarios
+				// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host
+				xfh, err := url.Parse(r.Header.Get("X-Forwarded-Host"))
```

Comment:
`X-Forwardef-Host` has the same format as `Host`, a hostname or IP address optionally followed by a colon and port, no?
Shouldn't it be parsed the same way as the `Host` header, using `util.SplitHostPortDefault` instead of `url.Parse` (where a scheme is required)?

Reply:
Thanks for your review @DasSkelett - I've seen the `X-Forwarded-Host` header value only as a plain hostname w/o protocol or port. That's also how it is described here: [X-Forwarded-Host](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host) 

But nevertheless, happy to change that and use `util.SplitHostPortDefault` instead. Testing it locally now.